src/de/mod.rs

```diff
@@ -461,34 +479,110 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         match peek {
             b'"' => {
                 self.eat_char();
-                visitor.visit_borrowed_str(self.parse_str()?)
+
+                let s = self.parse_str()?;
+
+                if s.as_bytes().contains(&b'\\') {
```

Comment:
It seems like we should avoid the de-escaping process if there's no de-escape buffer (i.e. user is not looking to de-escape the string).

Reply:
I'd say in that case we should return an error, as I'd consider returning encoded data a logic error. For the case where the user is not looking to de-escape strings, I was thinking as part of the design of having a struct along the line of the following:

```rust

enum UnescapedFragment<'a> {
    Str(&'a str),
    EscapedChar(c)
}

struct EscapedString<'a>(&'a str);

impl <'a> EscapedString<'a> {
    fn fragments(&self) -> impl Iterator<Item = Result<UnescapedFragment<'a>, SomeErrorType> {
        todo!()
    }
}

```

Reply:
I disagree. If the user hasn't opted-in to string escape management, we should just provide them the unescaped version to be the same as previous implementations. If a user was previously calling `from_slice()`, suddenly that code may begin breaking based on differing inputs. I would say we should set a flag in the deserializer to mark if we should be escaping characters.

Reply:
Ah does this work because the escaped string is moved after the de-escaping process? (i.e. into a heapless::String)

Reply:
The easiest was to have a flag in the deserializer is to have the string_unescape buffer be an Option, i.e.

```rust
pub struct Deserializer<'b, 's> {
    slice: &'b [u8],
    index: usize,
    string_unescape_buffer: Option<&'s mut [u8]>,
}
```

The when `serde::Deserializer::deserialize_str` is called, if `string_unescape_buffer` is `None`, it simply passes in the escaped string, which can a borrow of the input data, but if `string_unescape_buffer` is `Some(...)`, it attempts to deserialize the string with the current logic.

Reply:
Yeah that's what I was thinking as well :)